src/data/scripts/internet/main.py

```diff
@@ -170,33 +168,74 @@ def filter_field(data, source):
         data["updated_date"] = data["created_date"]
     return data
 
+if __name__ == "__main__":
+
+    if not os.path.exists(f"{output_dir}/{version}/data/"):
+        os.makedirs(f"{output_dir}/{version}/data/")
 
-NUM_PROC = int(args.num_proc)
-DATASET_TO_FILETYPE = {"mc4": "json"}
+    if scratch_location:
+        if not os.path.exists(f"{scratch_location}/{version}/data/"):
+            os.makedirs(f"{scratch_location}/{version}/data/")
+        scratch_writer = jsonlines.open(f"{scratch_location}/{version}/data/data.jsonl", "w")
+        
 
-if __name__ == "__main__":
-    with jsonlines.open(args.output_file, "w") as writer:
-        if args.source in DATASET_TO_FILETYPE.keys():
+    with jsonlines.open(f"{output_dir}/{version}/data/data.jsonl", "w") as writer:
+        
+        print("Loading dataset")
+
+        if source in DATASET_TO_FILETYPE.keys():
             dataset = load_dataset(
-                DATASET_TO_FILETYPE[args.source],
-                data_files=args.input_file,
-                split="train",
+                DATASET_TO_FILETYPE[source],
+                data_files=input_path,
             )
+
+        else:
+            dataset = load_from_disk(input_path)
+        
+        print(dataset)
+        if "train" in dataset.column_names:
+            dataset = dataset["train"]
+        if "id" not in dataset.column_names and "source_id" not in dataset.column_names:
             dataset = dataset.add_column(
                 "source_id", [i for i in range(len(dataset))]  # noqa: C416
             )
-        else:
-            dataset = load_from_disk(args.input_file)
 
+        print("Loaded dataset")
+        
         dataset = dataset.map(
             process_chunk_data,
             num_proc=NUM_PROC,
             batched=True,
-            batch_size=int(args.batch_size),
+            batch_size=int(batch_size),
+            # keep_in_memory=True,
+            cache_file_name = f"hf_cache/{source}/processed.arrow" # Incase that I cannot write in public_datasets, so I write in this instead
         )
 
-        if "train" in dataset.column_names:
-            dataset = dataset["train"]
-
         for data in dataset:
-            writer.write(filter_field(data, args.source))
+            
+            filtered_data = filter_field(data, source)
+            writer.write(filtered_data)
+            if scratch_location:
+                scratch_writer.write(filtered_data)
+        
+        print("Finish processing")
+
+        info = {"source":source, "current_version":version}
+        json.dump(info, open(f"{output_dir}/info.json","w"))
+
+        metadata = {"dataset_name":source, 
```

Comment:
@TODO (Me, Norapat): Will refactor logic about metadata into utils module so that others proprocessing code can also use it